### PR TITLE
Popover

### PR DESCRIPTION
**Pull request by jweisbeck**

### CODE REVIEW

**Comment by jweisbeck** [comment]
src/popover/index.js is the main directive file. It has a local service attached called BdSubmenu

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
   },
   "devDependencies": {
     "angular": ">=1.5",
+    "angular-mocks": "1.5.9",
```
Comment (jweisbeck) [inline package.json:33]: tests need this dep.

File: src/popover/index.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        </a>
+        <div class="popover__container popover__right" ng-class="ctrl.popoverContainerClass">
+            <div insert-point="popover-body"
+                 ng-class="{ 'pointer': ctrl.pointer }"
+                 ng-show="ctrl.isOpen"
+                 ng-mouseenter="ctrl.mouseEnter($event)"
+                 ng-mouseleave="ctrl.mouseLeave($event)">
+                <span class="popover__pointer" ng-class="ctrl.pointerClass" ng-if="::ctrl.pointer"></span>
+
+                <!-- anything within <popover-body> will appear here -->
```
Comment (digitil) [inline src/popover/index.js:74]: after this we should be able to refactor `insert-point` to be ngTransclude

File: src/popover/popover.test.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        $compile,
+        scope,
+        popoverCtrl,
+        mockSubmenuSvc,
+        mockPopUps,
+        mockFilter;
+
+    beforeEach(angular.mock.module('buildium.angular-elements.popover', function ($provide) {
+        mockSubmenuSvc = {
+            display : function display() {},
```
Comment (digitil) [inline src/popover/popover.test.js:19]: we create these noops and then spy on them later, but we can do that in one step if we set this to `jasmine.createSpy('display')` here, for example

**Review by digitil — APPROVED** [review]

File: src/popover/popover.test.js (hunk 3 above)
Comment (jweisbeck) [reply src/popover/popover.test.js:19]: good point.

**Pull request merged**